### PR TITLE
adds EncodeLike Slice for BoundedVec

### PR DESCRIPTION
**Pull request by ferrell-code**
Motivation is that it seems like `&[T]` should be able to read storage for key values of `BoundedVec<T, S>`

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1740969

File: primitives/runtime/src/bounded/bounded_vec.rs [hunk 1]
```diff
@@ -290,9 +290,12 @@ impl<T: Decode, S: Get<u32>> Decode for BoundedVec<T, S> {
 	}
 }
 
-// `BoundedVec`s encode to something which will always decode as a `Vec`.
+// `BoundedVec` encode to something which will always decode as a `Vec`.
 impl<T: Encode + Decode, S: Get<u32>> EncodeLike<Vec<T>> for BoundedVec<T, S> {}
 
+// `BoundedVec` encodes to something which will always decode as a `Slice`.
+impl<T: Encode + Decode, S: Get<u32>> EncodeLike<&[T]> for BoundedVec<T, S> {}
```
Comment (ggwpez) [inline primitives/runtime/src/bounded/bounded_vec.rs:297]: Could you please add a test?

File: primitives/runtime/src/bounded/bounded_vec.rs [hunk 2]
```diff
@@ -290,9 +290,12 @@ impl<T: Decode, S: Get<u32>> Decode for BoundedVec<T, S> {
 	}
 }
 
-// `BoundedVec`s encode to something which will always decode as a `Vec`.
+// `BoundedVec` encode to something which will always decode as a `Vec`.
 impl<T: Encode + Decode, S: Get<u32>> EncodeLike<Vec<T>> for BoundedVec<T, S> {}
 
+// `BoundedVec` encodes to something which will always decode as a `Slice`.
```
Comment (ferrell-code) [inline primitives/runtime/src/bounded/bounded_vec.rs:296]: Changed the comment cause you can't decode to slice as it isn't an owned value. As I understand all that matters is that they encode to the same thing

File: primitives/runtime/src/bounded/bounded_vec.rs [hunk 3]
```diff
@@ -290,9 +290,13 @@ impl<T: Decode, S: Get<u32>> Decode for BoundedVec<T, S> {
 	}
 }
 
-// `BoundedVec`s encode to something which will always decode as a `Vec`.
+// `BoundedVec` encodes to something which will always decode as a `Vec`.
 impl<T: Encode + Decode, S: Get<u32>> EncodeLike<Vec<T>> for BoundedVec<T, S> {}
 
+// `Slice` encodes to something which will always decode as a `BoundedVec` (will fail if it is
```
Comment (ferrell-code) [inline primitives/runtime/src/bounded/bounded_vec.rs:296]: I assume it is ok if the decoding is fallible

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1742231

**Comment by gavofyork** [comment]
I think this PR is wrong. It would allow oversize slices to be placed in a bounded vec storage item without any error (you would only get an error on decode, which we generally assume cannot happen). The proper way to do it is to convert the slice to a `BoundedSlice` and place that in storage, which guarantees that the stored data is valid at all times (since the `BoundedSlice` could not be converted into otherwise).

**Comment by ferrell-code** [comment]
This violates the entire point of `BoundedVec`

**Pull request closed without merging**